### DISCUSSION TITLE
Bitcoin API retuning same address for Ordinal and Payment types.

### DISCUSSION THREAD

**Discussion by lbowles** [phantom/docs#173] [Help & Support]
### Summary

This is for the Bitcoin wallet integration.

When using `await window.phantom.bitcoin.requestAccounts();` it returns 2 accounts:

**1**
- address: "bc1q7gk64fggs928h3z90gskku938hs3f9ghj24ef2"
- addressType:  "p2wpkh"
- publicKey: "0287b328721fa4f7d3fd16d78c69b4ff6513865076adea6d844cadca26fd38584d"
- purpose: "payment"
**2**
- address: "bc1q7gk64fggs928h3z90gskku938hs3f9ghj24ef2"
- addressType:  "p2wpkh"
- publicKey: "0287b328721fa4f7d3fd16d78c69b4ff6513865076adea6d844cadca26fd38584d"
- purpose: "ordinals"

The 2 accounts have the same type and public key when they should be different, I expect the "ordinals" type to be a taproot address and not segwit. Is there any way to get the correct address information for both types?

### Steps to Reproduce

`const accounts = await window.phantom.bitcoin.requestAccounts();
        console.log("Phantom connected addresses:", accounts);`
        
See more in the request accounts section in the [docs](https://docs.phantom.app/bitcoin/establishing-a-connection)

I have tested this with different wallets on Chrome and Firefox

### Phantom Version

24.0.1

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hey @lbowles,

Thank you for writing in!

I'm investigating and will follow up once I have more info.

**Reply by lbowles** [reply]
Hi, any update on this?

**Reply by adamdelphantom** [reply]
None, yet. I was able to reproduce. I will keep you updated. But please feel free to check in any time.

**Reply by adamdelphantom** [reply]
@lbowles Are you still seeing this issue w/ Phantom 24.4.0?

**Comment by kyleleow** [comment]
For anyone stumbled upon this, in phantom wallet browser extension click the account icon at top left to toggle the side menu > click settings icon > Preferences > Preferred Bitcoin Addresses > Enable "Taproot"
